eos-phone-home

```diff
@@ -277,6 +281,54 @@ class PhoneHome(object):
 
         return False
 
+    def _get_mac_hash(self):
+        mac_hash = None
+
+        try:
+            ifaces = []
+            wireless = []
+
+            for iface in glob.glob('/sys/class/net/*'):
+                # skip virtual interfaces
+                device = os.path.join(iface, 'device')
+                if not os.path.exists(device):
+                    continue
+
+                # skip USB devices
+                subsystem = os.readlink(os.path.join(device, 'subsystem'))
+                if os.path.basename(subsystem) == 'usb':
+                    continue
+
+                # build separate list of wireless interfaces
+                if os.path.exists(os.path.join(iface, 'wireless')):
+                    wireless.append(iface)
+                else:
+                    ifaces.append(iface)
+
+            # prefer the fixed interfaces if there are any, wireless if not
+            if len(ifaces) == 0:
+                ifaces = wireless
+
+            if len(ifaces) == 0:
+                return mac_hash
+
+            devs = {}
+            for iface in ifaces:
+                mac = open(os.path.join(iface, 'address')) \
```

Comment:
If you spell this as follows you avoid the nasty line continuation backslash:
```python
with open(os.path.join(iface, 'address')) as mac_file:
    mac = mac_file.read().strip().lower()
```

Reply:
I also had this before but I turned it into an exciting list generator comprehension thingy, before deciding to change it to a dict for decent debug strings, then deciding that was horrific for readibility.